File: frontend/src/dto/Preferences.ts
export interface Preferences {
  id: number | undefined;
  profileId: string;
  preferredCity: string | null;
  preferredCountry: string | null;
  animations: boolean;
  notifications: boolean;
}

export const defaultPreferences: Preferences = {
  id: undefined,
  profileId: "",
  notifications: true,
  preferredCity: null,
  preferredCountry: null,
  animations: true,
};

File: backend/src/main/java/com/cakmak/mondatelier/Model/Preferences.java
package com.cakmak.mondatelier.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Entity
@Table(name = "user_preferences")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Preferences {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long id;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "profile_id", unique = true)
    private Profile profile;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "language_id")
    private Language language;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "preferred_city_id")
    private City preferredCity;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "preferred_country_id")
    private Country preferredCountry;

    private Boolean animations;

    private Boolean notifications;
}
